src/services/widgets/widgets.spec.js

```diff
@@ -56,14 +56,105 @@ describe('<> ImpacWidgetsSvc', function () {
     xit('updates the widget\'s settings and eventually reload the widget');
   });
 
+  describe('#updateAllSameWidgets(:dashboard, :settings)', function () {
+    var dashboard, settings, paramName, updatedDashboard, anotherSettings;
+    beforeEach(function () {
+      spyOn(svc, 'load').and.returnValue($q.resolve(config));
+      dashboard = angular.copy(currentDhb);
+      settings = {
+        toMetadata: function () {
+          return {
+            status_selection: {
+              values: [],
+              reach: "dashboard"
+            }
+          }
+        }
+      };
+      anotherSettings = {
+        toMetadata: function () {
+          return {
+            another_status_selection: {
+              values: [],
+              reach: "dashboard"
+            }
+          }
+        }
+      };
+
+      dashboard.widgets[0].settings = [angular.copy(settings)];
+      dashboard.widgets[1].settings = [angular.copy(anotherSettings)];
+
+      paramName = _.keys(settings.toMetadata())[0];
+
+      dashboard[paramName] = {
```

Comment:
Are you using this later in the spec? I feel like testing that `DashboardsSvc.update` has been called with `dashboard` should be enough